libs/server/Objects/ObjectUtils.cs

```diff
@@ -100,7 +100,7 @@ public static unsafe bool ReadScanInput(byte* input, int length, ref SpanByteAnd
                     if (!RespReadUtils.ReadByteArrayWithLengthHeader(out var countParameterValue, ref input_currptr, input + length))
                         return false;
 
-                    int.TryParse(Encoding.ASCII.GetString(countParameterValue), out countInInput);
+                    _ = Utf8Parser.TryParse(countParameterValue, out countInInput, out _, default);
```

Comment:
**NOTE:** We should make sure we handle parsing errors gracefully, as this can have unintended consequences. I don't think we can handle this in this PR, so this only serves as a note for the future.

Reply:
Yeah, it hurts to leave them like this but I really can't address all the places where we swallow errors in this PR.